x-pack/legacy/plugins/ml/public/application/timeseriesexplorer/components/entity_control/entity_control.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiComboBox,
+  EuiComboBoxOptionProps,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiToolTip,
+} from '@elastic/eui';
+
+export interface Entity {
+  fieldName: string;
+  fieldValue: any;
+  fieldValues: any;
+}
+
+function getEntityControlOptions(entity: Entity): EuiComboBoxOptionProps[] {
+  if (!Array.isArray(entity.fieldValues)) {
+    return [];
+  }
+
+  return entity.fieldValues.map(value => {
+    return { label: value };
+  });
+}
+
+interface EntityControlProps {
+  entity: Entity;
+  entityFieldValueChanged: (entity: Entity, fieldValue: any) => void;
+  onSearchChange: (entity: Entity, queryTerm: string) => void;
+  forceSelection: boolean;
+}
+
+interface EntityContolState {
```

Comment:
Typo, should be `EntityControlState`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/53879/commits/c66af6e20807829e3fa8fcbb511d75cbb278d0dd